docs: update readme and add logging

leonardovida opened the pull request:
This PR:
- Updates the readme
- Adds some logging and typing

Still to do:
- Test cursor using `uvx` and add documentation around that

krish-adi reviewed (commented):
some comments on the readme and logging.

leonardovida commented:
<!-- linear:replyTo:2674041385 -->

Sorry for the mess, I removed the section on cursor and the arguments I added and didn't clean up the readme well enough. Will fix later this evening

leonardovida commented on src/mcp_server_motherduck/server.py:
```diff
@@ -70,22 +72,24 @@ def query(self, query: str) -> str:
                 # Duckbox version of the output
                 buffer = io.StringIO()
                 with redirect_stdout(buffer):
-                    self.conn.sql(query).show(max_rows=100, max_col_width=20)
```
@krish-adi these parameters don't exist? I tested them locally and they do not have any effect

krish-adi replied:
i'll check them out in my local to test it...

tdoehmen commented on src/mcp_server_motherduck/server.py:
```diff
@@ -20,7 +20,7 @@
 class DatabaseClient:
```
Can we update the server version here as well?
`SERVER_VERSION = "0.3.4"`

krish-adi reviewed (approved):
LGTM
